packages/admin/admin/src/mui/menu/CollapsibleItem.tsx

```diff
@@ -78,26 +79,29 @@ export const MenuCollapsibleItem = (inProps: MenuCollapsibleItemProps) => {
             // child is selected
             if (checkIfPathInLocation(child)) {
                 hasSelectedChild.current = true;
-                setIsSubmenuOpen(true);
             }
 
             // sub child is selected
             const subChildElements =
                 "children" in child.props ? Children.map(child?.props?.children, (child: MenuChild) => child) : ([] as MenuChild[]);
             if (subChildElements?.some((child: MenuChild) => child.props && checkIfPathInLocation(child))) {
                 hasSelectedChild.current = true;
-                setIsSubmenuOpen(true);
             }
 
             const newItemLevel = itemLevel + 1;
 
             return cloneElement<MenuCollapsibleItemProps | MenuItemRouterLinkProps | MenuItemProps>(child, {
                 level: newItemLevel === 1 || newItemLevel === 2 || newItemLevel === 3 ? newItemLevel : undefined,
                 isMenuOpen,
-                isCollapsibleOpen: isSubmenuOpen,
             });
         });
-    }, [children, isMenuOpen, isSubmenuOpen, itemLevel, location.pathname]);
+    }, [children, isMenuOpen, itemLevel, location.pathname]);
+
+    useEffect(() => {
+        if (hasSelectedChild.current) {
+            setIsSubmenuOpen(true);
```

Comment:
in my opinion `setIsSubmenuOpn` can be misleading and we have to separat this into two states

- `isSubmenuOpen` -> that's true when the submenu is really open.
- `isMenuSelected` or `isSubMenuSelected` -> that's true when the item itself, or a subitem is selected.